### PR TITLE
refactor(draft): refactoring draft action to edit action

### PR DESCRIPTION
**Pull request by ErwanSimonetti**
Draft action will be refactored to edit action, to be able to draft + undraft with a single `True` or `False` in the `.yml` file. Also, it will be possible to implement new edit actions related to the PR in the future.

Fixes MRGFY-616
Fixes MRGFY-619

### Development

- [X] All checks must pass (Semantic Pull Request, pep8, requirements, unit tests, functional tests, security checks, …)
- [X] The code changed/added as part of this pull request must be covered with tests
- [X] Features must have a link to a Linear task
- [X] Hotfixes must have a link to Sentry issue and the ``hotfix`` label

### Code Review

Code review policies are handled and automated by Mergify.

* When all tests pass, reviewers will be assigned automatically.
* When change is approved by at least one review and no pending review are
  remaining, pull request is retested against its base branch.
* The pull request is then merged automatically.

### CODE REVIEW

**Review by sileht — DISMISSED** [review]

File: mergify_engine/tests/functional/actions/test_edit.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        p, _ = await self.create_pr()
+        await self.run_engine()
+
+        pulls = await self.get_pulls()
+        self.assertEqual(1, len(pulls))
+
+        p = await self.get_pull(p["number"])
+        assert p["draft"] is True
+        rules["pull_request_rules"][0]["actions"]["edit"]["draft"] = False
+        print(f"rules = {rules}")
```
Comment (sileht) [inline mergify_engine/tests/functional/actions/test_edit.py:47]:
```suggestion
```

File: zfixtures/cassettes/TestCountSeats/test_count_seats/config.json [hunk 2]
```diff
@@ -1 +1 @@
-{"organization_id": 40527191, "organization_name": "mergifyio-testing", "repository_id": 399462964, "repository_name": "functional-testing-repo-ErwanSimonetti", "branch_prefix": "20211021090419"}
\ No newline at end of file
+{"organization_id": 40527191, "organization_name": "mergifyio-testing", "repository_id": 399462964, "repository_name": "functional-testing-repo-ErwanSimonetti", "branch_prefix": "20211021090419"}
```
Comment (sileht) [inline zfixtures/cassettes/TestCountSeats/test_count_seats/config.json:1]: You shouldn't have touched this test.

File: zfixtures/cassettes/TestCommentActionWithSub/test_comment_with_bot_account/config.json [hunk 3]
```diff
@@ -1 +1 @@
-{"organization_id": 40527191, "organization_name": "mergifyio-testing", "repository_id": 399462964, "repository_name": "functional-testing-repo-ErwanSimonetti", "branch_prefix": "20211014144507"}
\ No newline at end of file
+{"organization_id": 40527191, "organization_name": "mergifyio-testing", "repository_id": 399462964, "repository_name": "functional-testing-repo-ErwanSimonetti", "branch_prefix": "20211019145944"}
```
Comment (sileht) [inline zfixtures/cassettes/TestCommentActionWithSub/test_comment_with_bot_account/config.json:1]: You shouldn't have touched this test.

File: zfixtures/cassettes/TestCommentAction/test_comment_with_bot_account/config.json [hunk 4]
```diff
@@ -1 +1 @@
-{"organization_id": 40527191, "organization_name": "mergifyio-testing", "repository_id": 399462964, "repository_name": "functional-testing-repo-ErwanSimonetti", "branch_prefix": "20211014144708"}
\ No newline at end of file
+{"organization_id": 40527191, "organization_name": "mergifyio-testing", "repository_id": 399462964, "repository_name": "functional-testing-repo-ErwanSimonetti", "branch_prefix": "20211019150009"}
```
Comment (sileht) [inline zfixtures/cassettes/TestCommentAction/test_comment_with_bot_account/config.json:1]: You shouldn't have touched this test

**Review by sileht — CHANGES_REQUESTED** [review]
Except for the code duplication that needs to be refactored, and the unrelated fixtures changes, that looks good.

File: docs/source/actions/edit.rst [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+   :widths: 1 1 1 3
+
+   * - Key Name
+     - Value Type
+     - Default
+     - Value Description
+   * - ``draft``
+     - Boolean
+     - None
+     - If the pull request should be a Draft (True) or the other way around (False).
```
Comment (jd) [inline docs/source/actions/edit.rst:28]:
```suggestion
     - If the pull request should be a draft (``true``) or the other way around (``false``).
```

File: mergify_engine/tests/functional/actions/test_edit.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import yaml
+
+from mergify_engine.tests.functional import base
+
+
+class TestEditAction(base.FunctionalTestBase):
+    async def test_pr_to_draft_edit(self):
```
Comment (jd) [inline mergify_engine/tests/functional/actions/test_edit.py:23]: Can we have 2 tests, one for each state?

**Review by jd — DISMISSED** [review]

**Comment by jd** [comment]
release note?

File: mergify_engine/tests/functional/actions/test_edit.py (hunk 6 above)
Comment (ErwanSimonetti) [reply mergify_engine/tests/functional/actions/test_edit.py:23]: You think we should create the pull request as a draft directly and then check if it's correctly marked as `ready for review` in another test ?

Comment (jd) [reply mergify_engine/tests/functional/actions/test_edit.py:23]: You should have:
- test 1 that creates PR as a draft and converts it to ready to review
- test2 that creates PR and converts it to a draft

File: docs/source/actions/edit.rst [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+   :widths: 1 1 1 3
+
+   * - Key Name
+     - Value Type
+     - Default
+     - Value Description
+   * - ``draft``
+     - Boolean
+     - None
+     - If the pull request should be a Draft (``true``) or the other way around (``false``).
```
Comment (jd) [inline docs/source/actions/edit.rst:28]:
```suggestion
     - If the pull request should be a draft (``true``) or the other way around (``false``).
```

File: docs/source/actions/edit.rst [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+
+Examples
+--------
+
+📜 Converting a Pull Request to a Draft on Check Failure
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If any of your continuous integration checks fails, it might be worth converting
```
Comment (jd) [inline docs/source/actions/edit.rst:37]:
```suggestion
If any of your continuous integration checks fail, it might be worth converting
```

**Review by jd — CHANGES_REQUESTED** [review]

**Review by jd — APPROVED** [review]

**Pull request merged**